plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/ScaleDeploymentHandler.java

```diff
@@ -48,37 +65,114 @@
 	
 	@Override
 	public Object execute(ExecutionEvent event) throws ExecutionException {
-		IReplicationController rc = getSelectedElement(event, IReplicationController.class);
-		if(rc != null) {
-			scaleUsing(event, rc, rc.getName());
+		IResourceUIModel selected = getSelectedElement(event, IResourceUIModel.class);
+		if(selected == null) {
 			return null;
 		}
-		Deployment deployment = getSelectedElement(event, Deployment.class);
-		if(deployment != null) {
-			Collection<IResourceUIModel> rcs = deployment.getReplicationControllers();
-			if(!rcs.isEmpty()) {
-				//there should be only 1 per deployment, we'll assume this is true
-				rc = (IReplicationController) rcs.iterator().next().getResource();
-				scaleUsing(event, rc, deployment.getResource().getName());
+
+		String name = null; //Show in job name and dialog title.
+		Deployment deployment = null;
+		IReplicationController rc = null;
+
+		if(selected instanceof Deployment) {
+			deployment = (Deployment)selected;
+			rc = findReplicationController(deployment);
+			name = deployment.getResource().getName();
+		} else if(selected.getResource() instanceof IReplicationController) {
```

Comment:
You can compare its kind like `ResourceKind.ReplicationController.equals(selected.getResource().getKind())`